Examples/Algorithms/Geant4HepMC/src/EventRecording.cpp

```diff
@@ -73,14 +74,66 @@ ActsExamples::ProcessCode ActsExamples::EventRecording::execute(
     // Begin with the simulation
     m_runManager->BeamOn(1);
 
+    // Test if the event was aborted
+    if (SteppingAction::instance()->eventAborted()) {
+      continue;
+    }
+
+    // Set event start time
     HepMC3::GenEvent event = ActsExamples::EventAction::instance()->event();
-    HepMC3::FourVector shift(0., 0., 0., part.time());
+    HepMC3::FourVector shift(0., 0., 0., part.time() / Acts::UnitConstants::s);
```

Comment:
There is no time unit at all but I will remove the Acts time here to make it consistent

Reply:
I assume that means `c == 1` and the time unit is the same as the length unit? That still means it might have a different unit definition as Acts (which internally uses the base unit for length and time) and you need to scale it correctly.

Reply:
HepMC only uses a length and an energy unit but does not convert any values. This is just for the right interpretation of the given values. In this particular case it wouldn't even matter whether I divide by the time and multiply it when using it inside Acts or just leave it as it is.

Reply:
I looked through the rest of the code and saw that you are also setting HepMC four-momenta directly from Acts data further below. Anytime you set a HepMC record from Acts data you should have an explicit conversion factor. I fear, that the fact that it works at the moment is purely accidental. The Acts base units for length/time and energy/momentum are mm and GeV and match your HepMC base unit definition. However, the Acts base units can be easily changed and no code should rely on a particular base unit.

My suggestion is to always use the appropriate `1 / 1_mm` or `1 / 1_GeV` conversion factors from Acts native units to mm and GeV respectively. Since these are all known at compile time, the compiler should optimize them away if they end up being one.

Reply:
@FabianKlimpel The time scaling factor is still seconds. It should be `1 / 1_mm`.